src/core/pointcloud/qgseptdecoder.cpp

```diff
@@ -468,30 +471,32 @@ QgsPointCloudBlock *__decompressLaz( FileType &file, const QgsPointCloudAttribut
   float t = common::since( start );
   QgsDebugMsgLevel( QStringLiteral( "LAZ-PERF Read through the points in %1 seconds." ).arg( t ), 2 );
 #endif
-
-  return new QgsPointCloudBlock(
-           count,
-           requestedAttributes,
-           data
-         );
+  QgsPointCloudBlock *block = new QgsPointCloudBlock(
+    count,
+    requestedAttributes,
+    data, scale, offset
+  );
+  return block;
```

Comment:
Why do we create a new temporary variable `block` instead of just returning the new instance?